articles/clustering-in-unsupervised-ml/index.md

```diff
@@ -0,0 +1,157 @@
+3.	Use the Euclidean distance (between centroids and data points) to assign every data point to the closest cluster. 
+4.	Recalculate the centers of all clusters (as an average of the data points that have been assigned to each of them). 
+5.	Steps 3-4 should be repeated until there is no further change.
+
+![K-Mean Algorithm](/engineering-education/clustering-in-unsupervised-machine-learning/k-mean-algorithm.jpg)
+[Image Source: Tutorial Ride](https://www.tutorialride.com/data-mining/k-means-clustering-in-data-mining.htm)
+
+###### Advantages
++ It is very efficient in terms of computation
++ K-Means algorithms can be implemented easily
+
+###### Disadvantages
++ K-Means algorithms are not effective in identifying classes in groups that are spherically distributed.
++ The random selection of initial centroids may make some outputs (fixed training set) to be different. This may affect the entire algorithm process. 
+
+##### Hierarchical Clustering
+In this type of clustering, an algorithm is used to construct a hierarchy (of clusters). This algorithm will only end if there is only one cluster left. 
+
+Unlike K-means clustering, hierarchical clustering does not start by identifying the number of clusters. Instead, it starts by allocating each point of data to its cluster. 
+
+A dendrogram is a simple example of how hierarchical clustering works. 
+![Hierarchical Clustering Dendrogram](/engineering-education/clustering-in-unsupervised-machine-learning/hierarchical-clustering-dendrogram.png)
+[Image Source: Scikit-Learn](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_dendrogram_001.png)
+
+In the above diagram, the bottom observations that have been fused are similar, while the top observations are different. 
+
+###### Algorithm steps
+1.	Allocate each data point to its cluster. 
+2.	Use Euclidean distance to locate two closest clusters. These clusters should be merged to form one cluster.
+3.	Determine the distance between clusters that are near each other. The nearest clusters should be combined until all the data items have been grouped to form a single cluster.
+
+###### Advantages
++ The representations in the hierarchy provide meaningful information.
++ It does not require the specification of the number of clusters.
++ It is resourceful for the construction of dendrograms.
+
+###### Disadvantages
++ Hierarchical models have a high sensitivity to outliers. In the presence of outliers, the models do not perform well
++ The computation of hierarchical clustering is costly.
+
+##### Density-Based Spatial Clustering of Applications with Noise (DBSCAN)
+This is a density-based clustering that involves the grouping of data points that are close to each other. Data points that are far from each other are marked as outliers. Data is sorted based on commonalities. 
+
+###### Algorithm steps
+1.	In the first step, a core point should be identified. The core point radius is given as ε. Create a group  for each core point.

```

Comment:
What is the core point? Define core point.

Reply:
I have created a new section titled 'key concepts'. In this section, i have explained the main terms that have been used in the algorithm steps. These include MinPts, core point, border point and noise point.